File: schemas/media-streaming/convex/artist_genres.ts
// artist_genres: many-to-many relationship between artists and genres.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const artist_genres = defineTable({
  artistId: v.id("artists"),
  genreId: v.id("genres"),
})
  .index("by_artist_id_genre_id", ["artistId", "genreId"])
  .index("by_genre_id", ["genreId"]);

File: schemas/media-streaming/spacetimedb/artist_genres.rs
// artist_genres: Many-to-many association between artists and genres.
// See README.md for full design rationale.

#[spacetimedb::table(name = artist_genres, public)]
pub struct ArtistGenre {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub artist_id: String, // UUID — FK → artists.id (cascade delete)
    #[index(btree)]
    pub genre_id: String, // UUID — FK → genres.id (cascade delete)
}
// Composite unique: (artist_id, genre_id) — enforce in reducer logic
